src/extension/byok/vscode-node/azureProvider.ts

```diff
@@ -65,4 +69,89 @@ export class AzureBYOKModelProvider extends CustomOAIBYOKModelProvider {
 	protected override resolveUrl(modelId: string, url: string): string {
 		return resolveAzureUrl(modelId, url);
 	}
+
+	protected override async getModelsWithCredentials(silent: boolean): Promise<BYOKKnownModels> {
+		// Check user's authentication preference from settings github.copilot.chat.azureAuthType (default: 'entraId')
+		const authType = this._configurationService.getConfig(ConfigKey.AzureAuthType);
+
+		if (authType === 'entraId') {
```

Comment:
ideally store these constant strings in an enum class that stores EntraID key/s key-based auth options